src/test/java/com/github/tomakehurst/wiremock/SnapshotAcceptanceTest.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (C) 2017 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.tomakehurst.wiremock.common.Json;
+import com.github.tomakehurst.wiremock.testsupport.WireMockTestClient;
+import org.junit.After;
+import org.junit.Test;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static com.github.tomakehurst.wiremock.testsupport.TestHttpHeader.withHeader;
+import static com.github.tomakehurst.wiremock.testsupport.WireMatchers.equalToJson;
+import static junit.framework.TestCase.assertFalse;
+import static org.junit.Assert.assertThat;
+
+public class SnapshotAcceptanceTest extends AcceptanceTestBase {
+    private WireMockServer proxyingService;
+    private WireMockTestClient proxyingTestClient;
+
+    public void init() {
+        proxyingService = new WireMockServer(wireMockConfig()
+            .dynamicPort()
+            .withRootDirectory("src/test/resources/empty"));
+        proxyingService.start();
+        proxyingService.stubFor(proxyAllTo("http://localhost:" + wireMockServer.port()));
+
+        proxyingTestClient = new WireMockTestClient(proxyingService.port());
+        wireMockServer.stubFor(any(anyUrl()).willReturn(ok()));
+    }
+
+    @After
+    public void proxyServerShutdown() {
+        proxyingService.stop();
+    }
+
+    private static final String DEFAULT_SNAPSHOT_RESPONSE =
```

Comment:
Also, one issue with this change is that the order of the stub mappings will be lost, because JSON objects are unordered. This might be an issue is someone wants to do post-processing on the returned mappings in the order the request was received. Probably not a significant concern, but it's something to consider.